src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs

```diff
@@ -26,19 +32,126 @@ internal ModelOperationsCatalog(IHostEnvironment env)
             Explainability = new ExplainabilityTransforms(this);
         }
 
+        private void Save<TSource>(DataViewSchema schema, IDataLoader<TSource> model, Stream stream)
+        {
+            using (var rep = RepositoryWriter.CreateNew(stream))
+            {
+                ModelSaveContext.SaveModel(rep, model, null);
+                SaveInputSchema(schema, rep);
+                rep.Commit();
+            }
+        }
+
         /// <summary>
         /// Save the model to the stream.
         /// </summary>
         /// <param name="model">The trained model to be saved.</param>
         /// <param name="stream">A writeable, seekable stream to save to.</param>
-        public void Save(ITransformer model, Stream stream) => model.SaveTo(_env, stream);
+        public void Save<TSource>(IDataLoader<TSource> model, Stream stream)
+            => Save(model.GetOutputSchema(), model, stream);
+
+        /// <summary>
+        /// Save a transformer model and the loader used to create its input data to the stream.
+        /// </summary>
+        /// <param name="loader">The loader that was used to create data to train the model</param>
+        /// <param name="model">The trained model to be saved</param>
+        /// <param name="stream">A writeable, seekable stream to save to.</param>
+        public void Save<TSource>(IDataLoader<TSource> loader, ITransformer model, Stream stream) =>
+            Save(loader.GetOutputSchema(), new CompositeDataLoader<TSource, ITransformer>(loader, new TransformerChain<ITransformer>(model)), stream);
+
+        /// <summary>
+        /// Save a transformer model and the schema of the data that was used to train it to the stream.
+        /// </summary>
+        /// <param name="inputSchema">The schema of the input to the transformer.</param>
```

Comment:
>The schema of the input to the transformer [](start = 38, length = 42)

While I think having the argument is important to *encourage* people to save the input model, we should also allow them to explicitly not save the input in cases where it is disallowed. The goal is to lead them in the right direction -- the old ways were merely misleading, not always wrong, so I think we should allow this to be `null`.